packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -548,7 +549,11 @@ export default class CodeGen {
         // Cloning here is necessary because `this.replace()` is destructive, and we might use the
         // node later during static content optimization
         expression = doStructuredClone(expression);
-        walk(expression, {
+        // TODO [#3370]: when the template expression flag is removed, the
+        // ComplexExpression type should be redefined as an ESTree Node. Doing
+        // so when the flag is still in place results in a cascade of required
+        // type changes across the codebase.
+        walk(expression as Node, {
```

Comment:
This comment is copied from:

https://github.com/salesforce/lwc/blob/8a0f98fb42aa0db38d0148e402a7a743e3ad7e56/packages/%40lwc/template-compiler/src/shared/types.ts#L65-L69